### PR TITLE
ENT-6914 Upgrade to Gradle 7, Kotlin 1.8 and JDK 17 (Ent)

### PR DESCRIPTION
**Pull request by arshadm**
This PR upgrades the project for Corda 4.12

Major changes:
1. Gradle 7.6
2. Kotlin 1.8
3. JDK 17

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -252,53 +197,12 @@ allprojects {
     }
     tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).configureEach {
         kotlinOptions {
-            languageVersion = "1.2"
-            apiVersion = "1.2"
-            jvmTarget = VERSION_1_8
+            languageVersion = "1.8"
```
Comment (adelel1) [inline build.gradle:286]: 1.9?

File: gradle.properties [hunk 2]
```diff
@@ -1,31 +1,44 @@
...
 # this versionSuffix value sould be left blank for a release
 # all other times during normal development cycle this should be set to '-SNAPSHOT'
-versionSuffix=-SNAPSHOT
+baseVersion=4.12
+versionSuffix=SNAPSHOT
 
 # The version of Corda that the shell depends on
 # When creating a release build, this should be changed to point to a specific Corda version (e.g 4.9)
-cordaReleaseVersion=4.12-SNAPSHOT
+cordaOSReleaseVersion=4.12-202309-01
```
Comment (adelel1) [inline gradle.properties]: Shouldn't this be 4.12-HC01?

File: gradle.properties [hunk 3]
```diff
@@ -1,31 +1,44 @@
...
 cordaReleaseGroup=com.r3.corda
 cordaOpenSourceReleaseGroup=net.corda
-cordaPlatformVersion=11
+cordaPlatformVersion=14
 
-kotlinVersion=1.2.71
+kotlinVersion=1.9.0
 crashVersion=1.7.6
+sshdCoreVersion=2.10.0
+quasarVersion=0.9.0_r3-SNAPSHOT
```
Comment (adelel1) [inline gradle.properties]: Why SNAPSHOT?

File: build.gradle (hunk 1 above)
Comment (arshadm) [reply build.gradle:286]: Not at the moment, will upgrade once language version 1.9 has stabilised.

File: gradle.properties (hunk 2 above)
Comment (arshadm) [reply gradle.properties]: Updated to use HC01

File: gradle.properties (hunk 3 above)
Comment (arshadm) [reply gradle.properties]: Modified to use release version.

**Review by adelel1 — DISMISSED** [review]
LGTM
Leaving to build team to review Jenkins file.

**Review by wzur-r3 — DISMISSED** [review]
LGTM for Jenkinsfile changes

**Review by adelel1 — APPROVED** [review]

**Pull request merged**